engine/execution/state/delta/view.go

```diff
@@ -46,12 +46,20 @@ type SpockSnapshot struct {
 	SpockSecret []byte
 }
 
-func AlwaysEmptyGetRegisterFunc(owner, key string) (flow.RegisterValue, error) {
+func AlwaysEmptyGetRegisterFunc(
+	id flow.RegisterID,
+) (
+	flow.RegisterValue,
+	error,
+) {
 	return nil, nil
 }
 
 // NewView instantiates a new ledger view with the provided read function.
 func NewView(readFunc GetRegisterFunc) *View {
+	if readFunc == nil {
```

Comment:
this is only logic change in the PR.  it simplifies unittest setup in a bunch of places.